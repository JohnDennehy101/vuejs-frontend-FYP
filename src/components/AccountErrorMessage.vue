<template>
  <div class="form-control" v-show="toggle" id="error-sign-up-notice">
    <i class="fas fa-exclamation-triangle"></i>
    <p>{{ errorMessage }}</p>

    <span @click="hideErrorMessage">
      <i class="fas fa-times-circle"></i>
    </span>
  </div>
</template>

<script>
export default {
  props: {
    toggle: {
      type: Boolean,
    },
    errorMessage: {
      type: String,
    },
  },
  data() {
    return {
      display: this.toggle,
    };
  },

  methods: {
    hideErrorMessage() {
      this.$emit("hideErrorMessage", true);
    },
  },
};
</script>

<style scoped lang="scss">
#error-sign-up-notice {
  border-radius: 5px;
  background-color: #e57373;
  color: white;
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 0.5rem;

  p {
    margin-left: 0.5rem;
    margin-right: 0.5rem;
  }
  i {
    pointer-events: all;
  }
  .form-control {
    width: 80%;
    margin: 0.5rem auto;
  }
}
</style>
