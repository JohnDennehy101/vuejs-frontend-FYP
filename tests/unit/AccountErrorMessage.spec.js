import { mount, shallowMount } from "@vue/test-utils";
import AccountErrorMessage from "@/components/AccountErrorMessage.vue";

describe("AccountErrorMessage.vue", () => {
  it("should render correctly", () => {
    const wrapper = shallowMount(AccountErrorMessage);
    expect(wrapper).toMatchSnapshot();
  });

  it("checks errorMessage prop is rendered correctly", async () => {
    const wrapper = mount(AccountErrorMessage, {
      props: {
        errorMessage: "Error Message",
      },
    });

    expect(wrapper.find("p").text()).toEqual("Error Message");
  });

  it("checks event is emitted on click of x icon to hide error message", async () => {
    const wrapper = mount(AccountErrorMessage, {
      data() {
        return {
          userEmail: "test@gmail.com",
        };
      },
    });
    await wrapper.find("span").trigger("click");
    expect(wrapper.emitted()).toHaveProperty("hideErrorMessage");
  });
});
