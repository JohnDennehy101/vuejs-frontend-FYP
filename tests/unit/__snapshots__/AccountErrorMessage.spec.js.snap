// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountErrorMessage.vue should render correctly 1`] = `
VueWrapper {
  "__app": Object {
    "_component": Object {
      "name": "VTU_ROOT",
      "render": [Function],
    },
    "_container": <div
      data-v-app=""
    >
      <div
        class="form-control"
        id="error-sign-up-notice"
        style="display: none;"
      >
        <i
          class="fas fa-exclamation-triangle"
        />
        <p />
        <span>
          <i
            class="fas fa-times-circle"
          />
        </span>
      </div>
    </div>,
    "_context": Object {
      "app": [Circular],
      "components": Object {
        "transition": Object {
          "name": "transition",
        },
        "transition-group": Object {
          "name": "transition-group",
        },
      },
      "config": Object {
        "compilerOptions": Object {},
        "errorHandler": undefined,
        "globalProperties": Object {},
        "isNativeTag": [Function],
        "optionMergeStrategies": Object {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": Object {},
      "emitsCache": WeakMap {},
      "mixins": Array [
        Object {
          "beforeCreate": [Function],
        },
      ],
      "optionsCache": WeakMap {},
      "propsCache": WeakMap {},
      "provides": Object {},
      "reload": [Function],
    },
    "_instance": Object {
      "a": null,
      "accessCache": Object {},
      "appContext": Object {
        "app": [Circular],
        "components": Object {
          "transition": Object {
            "name": "transition",
          },
          "transition-group": Object {
            "name": "transition-group",
          },
        },
        "config": Object {
          "compilerOptions": Object {},
          "errorHandler": undefined,
          "globalProperties": Object {},
          "isNativeTag": [Function],
          "optionMergeStrategies": Object {},
          "performance": false,
          "warnHandler": undefined,
        },
        "directives": Object {},
        "emitsCache": WeakMap {},
        "mixins": Array [
          Object {
            "beforeCreate": [Function],
          },
        ],
        "optionsCache": WeakMap {},
        "propsCache": WeakMap {},
        "provides": Object {},
        "reload": [Function],
      },
      "asyncDep": null,
      "asyncResolved": false,
      "attrs": Object {},
      "bc": null,
      "bm": null,
      "bu": null,
      "bum": null,
      "c": null,
      "components": null,
      "ctx": Object {},
      "da": null,
      "data": Object {},
      "directives": null,
      "ec": null,
      "effect": ReactiveEffect {
        "active": true,
        "allowRecurse": true,
        "deps": Array [
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
        ],
        "fn": [Function],
        "onTrack": undefined,
        "onTrigger": undefined,
        "scheduler": [Function],
      },
      "emit": [Function],
      "emitsOptions": null,
      "emitted": null,
      "exposeProxy": null,
      "exposed": null,
      "inheritAttrs": undefined,
      "isDeactivated": false,
      "isMounted": true,
      "isUnmounted": false,
      "m": null,
      "next": null,
      "parent": null,
      "props": Object {},
      "propsDefaults": Object {},
      "propsOptions": Array [
        Object {},
        Array [],
      ],
      "provides": Object {},
      "proxy": Object {},
      "refs": Object {
        "VTU_COMPONENT": Object {
          "display": false,
          "errorMessage": undefined,
          "hasOwnProperty": [Function],
          "hideErrorMessage": [Function],
          "toggle": false,
        },
      },
      "render": [Function],
      "renderCache": Array [],
      "root": [Circular],
      "rtc": null,
      "rtg": null,
      "scope": EffectScope {
        "active": true,
        "cleanups": Array [],
        "effects": Array [
          ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": Array [
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "scheduler": [Function],
          },
        ],
      },
      "setupContext": null,
      "setupState": Object {},
      "slots": Object {},
      "sp": null,
      "subTree": Object {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": null,
        "children": null,
        "component": Object {
          "a": null,
          "accessCache": Object {
            "@@__IMMUTABLE_ITERABLE__@@": 0,
            "@@__IMMUTABLE_RECORD__@@": 0,
            "_isMockFunction": 0,
            "constructor": 0,
            "display": 2,
            "errorMessage": 3,
            "hasOwnProperty": 4,
            "hideErrorMessage": 4,
            "isVueInstance": 0,
            "toJSON": 0,
            "toggle": 3,
            Symbol(Symbol.toStringTag): 0,
          },
          "appContext": Object {
            "app": [Circular],
            "components": Object {
              "transition": Object {
                "name": "transition",
              },
              "transition-group": Object {
                "name": "transition-group",
              },
            },
            "config": Object {
              "compilerOptions": Object {},
              "errorHandler": undefined,
              "globalProperties": Object {},
              "isNativeTag": [Function],
              "optionMergeStrategies": Object {},
              "performance": false,
              "warnHandler": undefined,
            },
            "directives": Object {},
            "emitsCache": WeakMap {},
            "mixins": Array [
              Object {
                "beforeCreate": [Function],
              },
            ],
            "optionsCache": WeakMap {},
            "propsCache": WeakMap {},
            "provides": Object {},
            "reload": [Function],
          },
          "asyncDep": null,
          "asyncResolved": false,
          "attrs": Object {},
          "bc": null,
          "bm": null,
          "bu": null,
          "bum": null,
          "c": null,
          "components": Object {},
          "ctx": Object {
            "display": false,
            "errorMessage": undefined,
            "hasOwnProperty": [Function],
            "hideErrorMessage": [Function],
            "toggle": false,
          },
          "da": null,
          "data": Object {
            "display": false,
          },
          "directives": null,
          "ec": null,
          "effect": ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": Array [
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "scheduler": [Function],
          },
          "emit": [Function],
          "emitsOptions": null,
          "emitted": null,
          "exposeProxy": null,
          "exposed": null,
          "inheritAttrs": undefined,
          "isDeactivated": false,
          "isMounted": true,
          "isUnmounted": false,
          "m": null,
          "next": null,
          "parent": [Circular],
          "props": Object {
            "errorMessage": undefined,
            "toggle": false,
          },
          "propsDefaults": Object {},
          "propsOptions": Array [
            Object {
              "errorMessage": Object {
                "0": false,
                "1": true,
                "type": [Function],
              },
              "toggle": Object {
                "0": true,
                "1": true,
                "type": [Function],
              },
            },
            Array [
              "toggle",
            ],
          ],
          "provides": Object {},
          "proxy": Object {
            "display": false,
            "errorMessage": undefined,
            "hasOwnProperty": [Function],
            "hideErrorMessage": [Function],
            "toggle": false,
          },
          "refs": Object {},
          "render": [Function],
          "renderCache": Array [
            [Function],
          ],
          "root": [Circular],
          "rtc": null,
          "rtg": null,
          "scope": EffectScope {
            "active": true,
            "cleanups": Array [],
            "effects": Array [
              ReactiveEffect {
                "active": true,
                "allowRecurse": true,
                "deps": Array [
                  Set {
                    [Circular],
                  },
                  Set {
                    [Circular],
                  },
                ],
                "fn": [Function],
                "onTrack": undefined,
                "onTrigger": undefined,
                "scheduler": [Function],
              },
            ],
          },
          "setupContext": null,
          "setupState": Object {},
          "slots": Object {},
          "sp": null,
          "subTree": Object {
            "__v_isVNode": true,
            "__v_skip": true,
            "anchor": null,
            "appContext": null,
            "children": Array [
              Object {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": null,
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <i
                  class="fas fa-exclamation-triangle"
                />,
                "key": null,
                "patchFlag": -1,
                "props": Object {
                  "class": "fas fa-exclamation-triangle",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 1,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "i",
              },
              Object {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": "",
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <p />,
                "key": null,
                "patchFlag": 1,
                "props": null,
                "ref": null,
                "scopeId": null,
                "shapeFlag": 1,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "p",
              },
              Object {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": Array [
                  Object {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": null,
                    "appContext": null,
                    "children": null,
                    "component": null,
                    "dirs": null,
                    "dynamicChildren": null,
                    "dynamicProps": null,
                    "el": <i
                      class="fas fa-times-circle"
                    />,
                    "key": null,
                    "patchFlag": -1,
                    "props": Object {
                      "class": "fas fa-times-circle",
                    },
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 1,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": "i",
                  },
                ],
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <span>
                  <i
                    class="fas fa-times-circle"
                  />
                </span>,
                "key": null,
                "patchFlag": 0,
                "props": Object {
                  "onClick": [Function],
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 17,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "span",
              },
            ],
            "component": null,
            "dirs": Array [
              Object {
                "arg": undefined,
                "dir": Object {
                  "beforeMount": [Function],
                  "beforeUnmount": [Function],
                  "mounted": [Function],
                  "updated": [Function],
                },
                "instance": Object {
                  "display": false,
                  "errorMessage": undefined,
                  "hasOwnProperty": [Function],
                  "hideErrorMessage": [Function],
                  "toggle": false,
                },
                "modifiers": Object {},
                "oldValue": undefined,
                "value": false,
              },
            ],
            "dynamicChildren": Array [
              Object {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": "",
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <p />,
                "key": null,
                "patchFlag": 1,
                "props": null,
                "ref": null,
                "scopeId": null,
                "shapeFlag": 1,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "p",
              },
            ],
            "dynamicProps": null,
            "el": <div
              class="form-control"
              id="error-sign-up-notice"
              style="display: none;"
            >
              <i
                class="fas fa-exclamation-triangle"
              />
              <p />
              <span>
                <i
                  class="fas fa-times-circle"
                />
              </span>
            </div>,
            "key": null,
            "patchFlag": 512,
            "props": Object {
              "class": "form-control",
              "id": "error-sign-up-notice",
            },
            "ref": null,
            "scopeId": null,
            "shapeFlag": 17,
            "slotScopeIds": null,
            "ssContent": null,
            "ssFallback": null,
            "staticCount": 0,
            "suspense": null,
            "target": null,
            "targetAnchor": null,
            "transition": null,
            "type": "div",
          },
          "suspense": null,
          "suspenseId": 0,
          "type": Object {
            "components": Object {},
            "data": [Function],
            "methods": Object {
              "hideErrorMessage": [Function],
            },
            "props": Object {
              "errorMessage": Object {
                "0": false,
                "1": true,
                "type": [Function],
              },
              "toggle": Object {
                "0": true,
                "1": true,
                "type": [Function],
              },
            },
            "render": [Function],
          },
          "u": null,
          "uid": 1,
          "um": null,
          "update": [Function],
          "vnode": [Circular],
          "withProxy": null,
        },
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <div
          class="form-control"
          id="error-sign-up-notice"
          style="display: none;"
        >
          <i
            class="fas fa-exclamation-triangle"
          />
          <p />
          <span>
            <i
              class="fas fa-times-circle"
            />
          </span>
        </div>,
        "key": null,
        "patchFlag": 0,
        "props": Object {
          "ref": "VTU_COMPONENT",
        },
        "ref": Object {
          "f": false,
          "i": [Circular],
          "k": undefined,
          "r": "VTU_COMPONENT",
        },
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": Object {
          "components": Object {},
          "data": [Function],
          "methods": Object {
            "hideErrorMessage": [Function],
          },
          "props": Object {
            "errorMessage": Object {
              "0": false,
              "1": true,
              "type": [Function],
            },
            "toggle": Object {
              "0": true,
              "1": true,
              "type": [Function],
            },
          },
          "render": [Function],
        },
      },
      "suspense": null,
      "suspenseId": 0,
      "type": Object {
        "name": "VTU_ROOT",
        "render": [Function],
      },
      "u": null,
      "uid": 0,
      "um": null,
      "update": [Function],
      "vnode": Object {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": Object {
          "app": [Circular],
          "components": Object {
            "transition": Object {
              "name": "transition",
            },
            "transition-group": Object {
              "name": "transition-group",
            },
          },
          "config": Object {
            "compilerOptions": Object {},
            "errorHandler": undefined,
            "globalProperties": Object {},
            "isNativeTag": [Function],
            "optionMergeStrategies": Object {},
            "performance": false,
            "warnHandler": undefined,
          },
          "directives": Object {},
          "emitsCache": WeakMap {},
          "mixins": Array [
            Object {
              "beforeCreate": [Function],
            },
          ],
          "optionsCache": WeakMap {},
          "propsCache": WeakMap {},
          "provides": Object {},
          "reload": [Function],
        },
        "children": null,
        "component": [Circular],
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <div
          class="form-control"
          id="error-sign-up-notice"
          style="display: none;"
        >
          <i
            class="fas fa-exclamation-triangle"
          />
          <p />
          <span>
            <i
              class="fas fa-times-circle"
            />
          </span>
        </div>,
        "key": null,
        "patchFlag": 0,
        "props": null,
        "ref": null,
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": Object {
          "name": "VTU_ROOT",
          "render": [Function],
        },
      },
      "withProxy": null,
    },
    "_props": null,
    "_uid": 0,
    "component": [Function],
    "config": Object {
      "compilerOptions": Object {},
      "errorHandler": undefined,
      "globalProperties": Object {},
      "isNativeTag": [Function],
      "optionMergeStrategies": Object {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.2.26",
  },
  "__setProps": [Function],
  "componentVM": Object {
    "display": false,
    "errorMessage": undefined,
    "hasOwnProperty": [Function],
    "hideErrorMessage": [Function],
    "toggle": false,
  },
  "isDisabled": [Function],
  "rootVM": Object {},
  "wrapperElement": <div
    class="form-control"
    id="error-sign-up-notice"
    style="display: none;"
  >
    <i
      class="fas fa-exclamation-triangle"
    />
    <p />
    <span>
      <i
        class="fas fa-times-circle"
      />
    </span>
  </div>,
}
`;
